File: backend/app/schemas/asset.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class AssetCreate(BaseModel):
    collection_id: int
    collection_item_id: int | None = None

    serial_number: str | None = Field(default=None, max_length=150)

    asset_category: str = Field(max_length=100)
    manufacturer: str | None = Field(default=None, max_length=100)
    model: str | None = Field(default=None, max_length=150)
    description: str | None = None

    received_by: str | None = Field(default=None, max_length=150)
    receiving_notes: str | None = None

    notes: str | None = None


class AssetUpdate(BaseModel):
    serial_number: str | None = Field(default=None, max_length=150)

    asset_category: str = Field(default=None, max_length=100)
    manufacturer: str | None = Field(default=None, max_length=100)
    model: str | None = Field(default=None, max_length=150)
    description: str | None = None

    received_by: str | None = Field(default=None, max_length=150)
    receiving_notes: str | None = None

    notes: str | None = None


class AssetResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    asset_code: str
    serial_number: str | None

    collection_id: int
    collection_item_id: int | None

    asset_category: str
    manufacturer: str | None
    model: str | None
    description: str | None

    status: str

    received_at: datetime
    received_by: str | None
    receiving_notes: str | None

    data_wipe_status: str
    data_wipe_method: str | None
    data_wipe_date: datetime | None
    data_wipe_reference: str | None

    final_disposition: str | None
    notes: str | None

    created_at: datetime
    updated_at: datetime

File: frontend/src/types/asset.ts
export type Asset = {
  id: number
  asset_code: string
  serial_number: string | null

  collection_id: number
  collection_item_id: number | null

  asset_category: string
  manufacturer: string | null
  model: string | null
  description: string | null

  status: string

  received_at: string
  received_by: string | null
  receiving_notes: string | null

  data_wipe_status: string
  data_wipe_method: string | null
  data_wipe_date: string | null
  data_wipe_reference: string | null

  final_disposition: string | null
  notes: string | null

  created_at: string
  updated_at: string
}